llama_stack/distribution/server/auth_providers.py

```diff
@@ -162,18 +177,99 @@ async def close(self):
             self._client = None
 
 
+JWT_AUDIENCE = "llama-stack"
+
+
+class JWTAuthProviderConfig(BaseModel):
+    # The JWKS URI for collecting public keys
+    jwks_uri: str
+    algorithm: str = "RS256"
+    cache_ttl: int = 3600
+
+
+class JWTAuthProvider(AuthProvider):
+    """
+    JWT token authentication provider that validates a JWT token and extracts access attributes.
+
+    This should be the standard authentication provider for most use cases.
+    """
+
+    def __init__(self, config: JWTAuthProviderConfig):
+        self.config = config
+        self._jwks_at: float = 0.0
+        self._jwks: dict[str, str] = {}
+
+    async def validate_token(self, token: str, scope: dict | None = None) -> TokenValidationResult:
+        """Validate a token using the JWT token."""
+        await self._refresh_jwks()
+
+        try:
+            kid = jwt.get_unverified_header(token)["kid"]
+            if kid not in self._jwks:
+                raise ValueError(f"Unknown key ID: {kid}")
+            key = self._jwks[kid]
+            claims = jwt.decode(
+                token,
+                key,
+                algorithms=[self.config.algorithm],
```

Comment:
The algorithm is also specified in the token header (key "alg"), taking it from there would mean that a service using something different should also work.